### PR TITLE
Watchtower pickup fixing

### PR DESCRIPTION
**Pull request by MrSmig**
Fix Watchtower's becoming immovable if you try to pick them up while they're clumped up together.

### CODE REVIEW

File: gamemodes/horde/entities/entities/horde_watchtower/init.lua [hunk 1]
```diff
@@ -68,7 +67,9 @@ function ENT:Think()
 end
 
 hook.Add("PlayerUse", "PickUpWatchtower", function(ply, ent)
-	if HORDE:IsWatchTower(ent) and ent:GetNWEntity("HordeOwner"):IsValid() and ent:GetNWEntity("HordeOwner") == ply then
+	if not HORDE:IsWatchTower(ent) then return false end
+    if not ent:GetNWEntity("HordeOwner"):IsValid() then return false end
```
Comment (wrefgtzweve) [inline gamemodes/horde/entities/entities/horde_watchtower/init.lua:71]: you're using tab and space indentation, please use one type of indentation in the entire file

in this case use tabs as horde uses them

File: gamemodes/horde/entities/entities/horde_watchtower/init.lua [hunk 2]
```diff
@@ -83,13 +84,27 @@ hook.Add("PlayerUse", "PickUpWatchtower", function(ply, ent)
                 end
             end
         end
+        if not ply.IsHoldingObject then
         local p = ent:GetPos()
```
Comment (wrefgtzweve) [inline gamemodes/horde/entities/entities/horde_watchtower/init.lua:88]: this should be indented

File: gamemodes/horde/entities/entities/horde_watchtower/init.lua [hunk 3]
```diff
@@ -83,13 +84,27 @@ hook.Add("PlayerUse", "PickUpWatchtower", function(ply, ent)
...
+        end
         ent.Horde_WatchtowerPickedUp = ply
         ent.Horde_WatchtowerPickupCd = CurTime() + 0.5
-    end
-end )
\ No newline at end of file
+end )
+
+hook.Add("OnPlayerPhysicsPickup","DetectPickup", function(ply, ent)
+
```
Comment (wrefgtzweve) [inline gamemodes/horde/entities/entities/horde_watchtower/init.lua:100]: no whitespace needed here

**Review by wrefgtzweve — CHANGES_REQUESTED** [review]

File: gamemodes/horde/entities/entities/horde_watchtower/init.lua (hunk 1 above)
Comment (MrSmig) [reply gamemodes/horde/entities/entities/horde_watchtower/init.lua]: Done

File: gamemodes/horde/entities/entities/horde_watchtower/init.lua (hunk 2 above)
Comment (MrSmig) [reply gamemodes/horde/entities/entities/horde_watchtower/init.lua]: wasn't sure if you were referring to the line itself or the block of code it surrounds so I indented the block

File: gamemodes/horde/entities/entities/horde_watchtower/init.lua [hunk 4]
```diff
@@ -4,92 +4,103 @@ include("shared.lua")
...
-        ply:PickupObject(ent)
-        ent.Horde_WatchtowerPickedUp = ply
-        ent.Horde_WatchtowerPickupCd = CurTime() + 0.5
-    end
-end )
\ No newline at end of file
+	if not HORDE:IsWatchTower(ent) then return false end
+	if not ent:GetNWEntity("HordeOwner"):IsValid() then return false end
+	if not ent:GetNWEntity("HordeOwner") == ply then return false end
+		if not ent.Horde_WatchtowerPickupCd then
```
Comment (wrefgtzweve) [inline gamemodes/horde/entities/entities/horde_watchtower/init.lua:73]: this shouldn't be indented

Comment (MrSmig) [reply gamemodes/horde/entities/entities/horde_watchtower/init.lua]: done

File: gamemodes/horde/entities/entities/horde_watchtower/init.lua [hunk 5]
```diff
@@ -4,92 +4,103 @@ include("shared.lua")
...
+	ent.Horde_WatchtowerPickedUp = ply
+	ent.Horde_WatchtowerPickupCd = CurTime() + 0.5
+end )
+
+hook.Add("OnPlayerPhysicsPickup","DetectPickup", function(ply, ent)
+	ply.IsHoldingObject = true
+end)
+
+hook.Add("OnPlayerPhysicsDrop","Detectdrop", function(ply, ent)
+	ply.IsHoldingObject = false
```
Comment (wrefgtzweve) [inline gamemodes/horde/entities/entities/horde_watchtower/init.lua:104]: this should be set to nil instead of false

**Pull request merged**